src/main/java/com/sourcery/gymapp/backend/authentication/service/CustomOidcUserService.java

```diff
@@ -0,0 +1,37 @@
+package com.sourcery.gymapp.backend.authentication.service;
+
+import com.sourcery.gymapp.backend.authentication.model.OidcUserAttributes;
+import com.sourcery.gymapp.backend.authentication.model.User;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserRequest;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserService;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.support.TransactionTemplate;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class CustomOidcUserService extends OidcUserService {
+
+    private final OidcUserAttributesExtractor attributesExtractor;
+    private final OidcUserProcessorService userProcessorService;
+    private final OidcUserEventService userEventService;
```

Comment:
Again if there is no constraints from the library, we can decrease number of services for these features
You could merge some of them, e.g. OidcUserEventService  into OidcUserProcessorService,
In my opinion there may be event only 2 services, but the choice is yours